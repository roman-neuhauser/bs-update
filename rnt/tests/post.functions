cleanup()
{
  rm -f "$PWD/test/exit.actual"
  rm -f "$PWD/test/err.actual"
  rm -f "$PWD/test/out.actual"
  rm -f "$PWD/test/exit.diff"
  rm -f "$PWD/test/err.diff"
  rm -f "$PWD/test/out.diff"
}

assert_fs()
{
  local ex=1 label=FAIL neg p t=-e tt

  p="${1:?}"
  if test $# -gt 1; then
    t="${1:?}" p="${2:?}"
  fi
  tt="$t"
  case "$t" in
  !*) neg='!' t="-${t#!}" ;;
  esac

  if test $neg "$t" "$p"; then
    ex=0
    label=good
  fi
  printf "%s: %s %s\n" $label "$tt" "$p"
  return $ex
}

assert_same()
{
  diff -u "${1:?}" "${2:?}"
}

