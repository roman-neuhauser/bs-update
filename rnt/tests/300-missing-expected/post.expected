good: -f test/out.actual
good: -f test/err.actual
good: -f test/exit.actual
good: -f test/stray.actual
good: !e test/out.diff
good: !e test/err.diff
good: -f test/exit.diff
good: -f test/stray.diff
good: !e test/out.diff.tmp
good: !e test/err.diff.tmp
good: !e test/exit.diff.tmp
good: !e test/stray.diff.tmp
